src/js/lib/monster.ui.js

```diff
@@ -2120,97 +2120,149 @@ define(function(require) {
+			}
+
+			var $container = _.get(options, 'container');
+			if (!_.isUndefined($container) && !($container instanceof $)) {
+				throw new TypeError('"options.container" is not a jQuery object');
+			}
+
+			var regexes = _.map([
+				{
+					key: 'strong',
+					regex: new RegExp('^(?=.{8,})(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[\\W_]).*$'),
+					color: '#18b309',
+					size: 100,
+					emojiChar: '&#x1F60E;'
+				},
+				{
+					key: 'good',
+					regex: new RegExp('^(?=.{8,})(((?=.*[A-Z])(?=.*[\\W_]))|((?=.*[A-Z])(?=.*[0-9]))|((?=.*[\\W_])(?=.*[0-9]))).*$'),
+					color: '#33db24',
+					size: 70,
+					emojiChar: '&#x1F603;'
+				},
+				{
+					key: 'medium',
+					regex: new RegExp('^(?=.{6,})((?=.*[\\W_])|(?=.*[A-Z])|(?=.*[0-9])).*$'),
+					color: '#ffcc33',
+					size: 50,
+					emojiChar: '&#x1F610;'
+				},
+				{
+					key: 'weak',
+					regex: new RegExp('^(?=.{6,}).*$'),
+					color: '#ff6a57',
+					size: 40,
+					emojiChar: '&#x1F61E;'
+				},
+				{
+					key: 'bad',
+					regex: new RegExp('^.+$'),
+					color: '#ff3d24',
+					size: 20,
+					emojiChar: '&#x1F621;'
+				},
+				{
+					key: 'empty',
+					regex: new RegExp('^\\s*$'),
+					color: '#c0c0c9',
+					size: 0,
+					emojiChar: '&nbsp;'
+				}
+			], function(val) {
+				return _.merge({
+					label: monster.util.tryI18n(i18n.passwordStrength, val.key)
+				}, val);
+			});
+			var templateName = 'monster-password-strength-' + display;
+			var $template = $(monster.template(
+				monster.apps.core,
+				templateName,
+				{
+					tooltipPosition: tooltipPosition
+				}));
+			var updateIndicator = _.get({
+				bar: function(strengthArgs) {
+					$template
+						.find('.monster-password-strength-bar')
+						.css({
+							backgroundColor: strengthArgs.color,
+							width: strengthArgs.size + '%'
 						});
-						break;
-					}
+					$template
+						.find('.monster-password-strength-label')
+						.html(strengthArgs.label);
+				},
+				emoji: function(strengthArgs) {
+					$template
+						.find('.monster-password-strength-icon')
+						.css('color', strengthArgs.color)
+						.html(strengthArgs.emojiChar);
+					$template
+						.find('.monster-password-strength-label')
+						.html(strengthArgs.label);
+				},
+				icon: function(strengthArgs) {
+					$template
+						.find('.monster-password-strength-icon')
+						.css('color', strengthArgs.color)
+						.attr('data-original-title', strengthArgs.label)
+						.tooltip('fixTitle');
 				}
+			}, display);
+			var checkStrength = function() {
+				_.each(regexes, function(strengthData) {
+					if (!strengthData.regex.test(input.val())) {
+						return;
+					}
 
-				if (options.container) {
-					options.container.append(strengthDisplay);
-				} else {
-					input.after(strengthDisplay);
-				}
+					updateIndicator(strengthData);
+					return false;
+				});
```

Comment:
This logic doesn't seem as straightforward as it could be, why not use `lodash#find` instead and check against `undefined`?